src/main/webapp/features/shared/services/game/Game.service.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+angular.module('catan')
+        .factory('Game', ['GameModel', '$q', 'Remote', function (GameModel, $q, Remote) {
+
+            var Game = function(idOrDetails) {
+                return new GameModel(idOrDetails);
+            };
+
+            Game.findPublic = function() {
+                return this.findByRemoteService("game", "listPublic");
+            };
+
+            Game.findCurrent = function() {
+                return this.findByRemoteService("game", "listCurrent");
+            };
+
+            Game.findByRemoteService = function(remoteGroup, remoteRequest) {
```

Comment:
this can be simplified, to pass only scope.typeOfGames from ctGamesList directive, and in such case we can remove this logic: (scope.typeOfGames === 'CURRENT') ? 'findCurrent' : 'findPublic'
and remove 2 methods from this service: 'findPublic' and 'findCurrent' , so that we can leave only universal method that takes typeOfGames as a parameter.